commit f1fa27f590e56c08bc1824974c5f69610133003d
Author: Gustavo A. R. Silva <gustavoars@kernel.org>
Date:   Wed Jul 22 13:58:52 2020 -0500

    net: qed_hsi.h: Avoid the use of one-element array
    
    One-element arrays are being deprecated[1]. Replace the one-element
    array with a simple value type '__le32 reserved1'[2], once it seems
    this is just a placeholder for alignment.
    
    [1] https://github.com/KSPP/linux/issues/79
    [2] https://github.com/KSPP/linux/issues/86
    
    Tested-by: kernel test robot <lkp@intel.com>
    Link: https://github.com/GustavoARSilva/linux-hardening/blob/master/cii/0-day/qed_hsi-20200718.md
    Signed-off-by: Gustavo A. R. Silva <gustavoars@kernel.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/drivers/net/ethernet/qlogic/qed/qed_hsi.h b/drivers/net/ethernet/qlogic/qed/qed_hsi.h
index 1af3f65ab862..559df9f4d656 100644
--- a/drivers/net/ethernet/qlogic/qed/qed_hsi.h
+++ b/drivers/net/ethernet/qlogic/qed/qed_hsi.h
@@ -361,7 +361,7 @@ struct core_tx_update_ramrod_data {
 	u8 update_qm_pq_id_flg;
 	u8 reserved0;
 	__le16 qm_pq_id;
-	__le32 reserved1[1];
+	__le32 reserved1;
 };
 
 /* Enum flag for what type of dcb data to update */